tests/test_views.py

```diff
@@ -788,3 +801,100 @@ def test_new_group_created(
                 kwargs=dict(group_pk=group.pk),
             ),
         )
+
+
+class TestGetNextRdfProjectId:
+    """Tests for the get_next_rdf_project_id utility function."""
+
+    def test_next_id(self, pi_project, rdf_allocation_dependencies):
+        """Check correct next id is generated from existing id values."""
+        rdf_resource = Resource.objects.get(name="RDF Project Storage Space")
+        rdf_id_attribute_type = AllocationAttributeType.objects.get(
+            name="RDF Project ID"
+        )
+
+        allocation_active_status = AllocationStatusChoice.objects.get(name="Active")
+        allocation = Allocation.objects.create(
+            project=pi_project, status=allocation_active_status
+        )
+        allocation.resources.add(rdf_resource)
+
+        last_project_number = 23
+        AllocationAttribute.objects.create(
+            allocation_attribute_type=rdf_id_attribute_type,
+            allocation=allocation,
+            value=format_project_number_to_id(last_project_number),
+        )
+
+        assert get_next_rdf_project_id() == format_project_number_to_id(
+            last_project_number + 1
+        )
+
+    def test_first(self, db):
+        """Check initial id generation."""
+        assert get_next_rdf_project_id() == format_project_number_to_id(1)
+
+
+class TestAddRDFStorageAllocation(LoginRequiredMixin):
+    """Tests for the add_rdf_storage_allocation view."""
+
+    def _get_url(self):
+        return reverse("imperial_coldfront_plugin:add_rdf_storage_allocation")
+
+    def test_non_admin_forbidden(self, user_member_or_manager, auth_client_factory):
+        """Test non-admin users cannot access the page."""
+        client = auth_client_factory(user_member_or_manager)
+        response = client.get(self._get_url())
+        assert response.status_code == HTTPStatus.FORBIDDEN
+
+    def test_get(self, superuser_client):
+        """Check form rendering."""
+        response = superuser_client.get(self._get_url())
+        assert response.status_code == HTTPStatus.OK
+        assert isinstance(response.context["form"], RDFAllocationForm)
+
+    @patch("imperial_coldfront_plugin.views.ldap_create_group_in_background")
+    def test_post(
+        self,
+        ldap_create_group_mock,
+        pi_project,
+        superuser_client,
+        rdf_allocation_dependencies,
+    ):
+        """Test successful project creation."""
+        end_date = timezone.datetime.max.date()
+        size = 10
+        response = superuser_client.post(
+            self._get_url(),
+            data=dict(project=pi_project.pk, end_date=end_date, size=size),
+        )
+        assertRedirects(response, reverse("home"), fetch_redirect_response=False)
+
+        project_id = format_project_number_to_id(1)
+        from coldfront.core.allocation.models import (
+            Allocation,
+            AllocationAttribute,
+            AllocationUser,
+        )
```

Comment:
Unless there's a reason for it, move these import to the top of the function. I'm surprised that `ruff` didn't freakout with this...

Reply:
Huh. Yeah, why didn't ruff catch that...?